Sources/VitaminUIKit/Components/Snackbar/VitaminSnackbar.swift

```diff
@@ -0,0 +1,360 @@
+        applyNewDismissOnTap()
+        applyNewTexts()
+        applyNewImage()
+        if autoDismiss {
+            launchAutoDismissal()
+        }
+    }
+}
+
+// MARK: - private methods for applying changes
+extension VitaminSnackbar {
+    // set up common layout constraints without activating them
+    private func setupCommonConstraints() {
+        commonConstraints.append(titleLabel.topAnchor.constraint(equalTo: self.topAnchor, constant: topBottomMargin))
+        commonConstraints.append(
+            messageLabel.bottomAnchor.constraint(equalTo: self.bottomAnchor, constant: -topBottomMargin)
+        )
+        commonConstraints.append(titleLabel.widthAnchor.constraint(equalTo: messageLabel.widthAnchor))
+        commonConstraints.append(titleLabel.bottomAnchor.constraint(equalTo: messageLabel.topAnchor))
+        commonConstraints.append(widthAnchor.constraint(lessThanOrEqualToConstant: maxWidth))
+    }
+
+    // set up layout constraints for image case without activating them
+    private func setupWithImageConstraints() {
+        withImageConstraints.append(
+            titleLabel.leftAnchor.constraint(equalTo: icon.rightAnchor, constant: imageLeftAndRightMargin)
+        )
+        withImageConstraints.append(messageLabel.leftAnchor.constraint(equalTo: titleLabel.leftAnchor))
+        withImageConstraints.append(icon.widthAnchor.constraint(equalToConstant: imageWidthAndHeight))
+        withImageConstraints.append(icon.heightAnchor.constraint(equalToConstant: imageWidthAndHeight))
+        withImageConstraints.append(
+            icon.leftAnchor.constraint(equalTo: self.leftAnchor, constant: imageLeftAndRightMargin)
+        )
+        withImageConstraints.append(icon.centerYAnchor.constraint(equalTo: self.centerYAnchor))
+        withImageConstraints.append(
+            messageLabel.rightAnchor.constraint(equalTo: self.rightAnchor, constant: -rightMargin)
+        )
+    }
+
+    // set up layout constraints for no image case without activating them
+    private func setupWithoutImageConstraints() {
+        withoutImageConstraints.append(titleLabel.centerXAnchor.constraint(equalTo: self.centerXAnchor))
+        withoutImageConstraints.append(messageLabel.centerXAnchor.constraint(equalTo: self.centerXAnchor))
+        withoutImageConstraints.append(
+            messageLabel.widthAnchor.constraint(lessThanOrEqualTo: self.widthAnchor, constant: -2 * rightMargin)
+        )
+    }
+
+    // apply changes when title or message are changed
+    private func applyNewTexts() {
+        titleLabel.attributedText = title.styled(as: .footnoteBold, with: VitaminColor.Core.Content.primary)
+        messageLabel.attributedText = message.styled(as: .footnote, with: VitaminColor.Core.Content.tertiary)
+    }
+
+    // apply changes when an image is set or changed
+    private func applyNewImage() {
+        if let image = image {
+            icon.image = image
+            if self.subviews.first(where: { $0 == icon }) == nil {
```

Comment:
Not sure but I think that we can do that.
```suggestion
            if icon.superview == nil {
```

Reply:
ok